src/librustc_span/hygiene.rs

```diff
@@ -732,12 +733,17 @@ pub enum ExpnKind {
 }
 
 impl ExpnKind {
-    pub fn descr(&self) -> Symbol {
+    pub fn descr(&self) -> Cow<'static, str> {
         match *self {
-            ExpnKind::Root => kw::PathRoot,
-            ExpnKind::Macro(_, descr) => descr,
-            ExpnKind::AstPass(kind) => Symbol::intern(kind.descr()),
-            ExpnKind::Desugaring(kind) => Symbol::intern(kind.descr()),
+            // FIXME(eddyb) is this even a good description for `Root`?
```

Comment:
It's ok, it shouldn't be user-visible and only appears in debug messages at most.